### DISCUSSION TITLE
[Unstable] Wreck/Beacon layer system is quite buggy in current state

### DISCUSSION THREAD

**Discussion by PanmanS** [FakeFishGames/Barotrauma#13730] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

This new system opens tons of creative paths for wreck/beacon making but it has some issues.
![Barotrauma_txonU5xxYn](https://github.com/FakeFishGames/Barotrauma/assets/42521948/cd89188b-a404-49df-a308-6cfd989b5935)

First: walls that are not activated by events still provide collision regardless of their levels being visible or not
![image](https://github.com/FakeFishGames/Barotrauma/assets/42521948/08d9ae78-2cbe-4536-a62f-3805ee991381)
![image](https://github.com/FakeFishGames/Barotrauma/assets/42521948/6ae57aea-ab51-4467-833e-8f9cd6d4819e)
![image](https://github.com/FakeFishGames/Barotrauma/assets/42521948/1b93b7b7-c963-4988-afc8-df39d1bb9707)

Second: Disabling a group with light sources removes lamp sprites, but not lights themselves. (self explanatory)

Third: it doesnt work properly with dropped items

And finally: it doesnt work properly with thalamus because it thalamus makes all items related to it show regardless of layer system.

### Reproduction steps

1. Create custom event for your wreck/beacon. Easiest option is just altering bandit even or doing it like in example below:
![image](https://github.com/FakeFishGames/Barotrauma/assets/42521948/448e8b14-7fad-4b34-881b-68a6c0b37d60)

2. Create layers that would be affected by event above.

3. finally load through `leveleditor` or `quickstart` and start testing using events above

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.4.0.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by ubertpendragon** [comment]
It just doesnt handle hiding colliders everything else is fine

**Comment by Regalis11** [comment]
Thank you for the report!

I would not actually consider this a bug, nor specific to the unstable build: this is how the group/layer system has always worked. We haven't added support for disabling colliders entirely, because it could easily lead to issues, for example when a collider becomes active while a character or a submarine is inside it. I could see it being useful though, so we might look into it at some point, but I'd consider it a feature request, so I'll transfer this discussion to the "ideas and suggestions" section.

**Reply by JeroonBosch** [reply]
> everything else is fine

It's actually still a bit work-in-progress, for example disabled spawnpoints are still able to be tagged and spawned on (and there might be other tweaks to it still), it will likely be improved upon in later unstables / later updates

**Reply by PanmanS** [reply]
Well that would be amazing, because it opens doors to making multiple wrecks in 1 file, without needing to dillute pool of em with morbillion versions of 1 sub
